File: javascript/pangram/pangram.js
function Pangram(word) {
    this.letters = word
        .replace(/[^a-z]/gi, '')
        .toLowerCase()
        .split('')
        .sort()
        .reduce(removeRepeats, []);
};

Pangram.prototype.isPangram = function() {
    return this.letters.length === 26;
};

function removeRepeats(acc, letter) {
    if (!acc.includes(letter)) {
        acc.push(letter);
    }
    return acc;
}

module.exports = Pangram;

File: haskell/pangram/src/Pangram.hs
module Pangram (isPangram) where

import Data.Char (isAlpha, toLower)
import Data.List (sort, group)

isPangram :: String -> Bool
isPangram text = total >= 26
    where
        letters = filter isAlpha $ map toLower text
        total = length $ group $ sort letters 
